go/extension/extension.go

```diff
@@ -141,6 +147,41 @@ func (e *FractalExt) GrantsFor(ctx *types.ExecutionContext, values ...*types.Sca
 	return encodeScalarValues(exist)
 }
 
+func (e *FractalExt) HexToBase58(ctx *types.ExecutionContext, values ...*types.ScalarValue) ([]*types.ScalarValue, error) {
+	inputHex, err := values[0].String()
+	if err != nil {
+		return nil, fmt.Errorf("convert value to string failed: %w", err)
+	}
+	binaryString, _ := hex.DecodeString(inputHex)
+	base58 := base58.Encode(binaryString)
+	var public_key string
+	if base58 == "" {
+		public_key = ""
+	} else {
+		public_key = fmt.Sprintf("ed25519:%s", base58)
+	}
+
+	return encodeScalarValues(public_key)
+}
+
+// This has very dumb logic: eth address returns EVM type, and NEAR returns otherwise.
+// TODO: make the logic more detailed and return error is the address is neither EVM no NEAR.
+func (e *FractalExt) DetermineWalletType(ctx *types.ExecutionContext, values ...*types.ScalarValue) ([]*types.ScalarValue, error) {
+	address, err := values[0].String()
+	if err != nil {
+		return nil, fmt.Errorf("convert value to string failed: %w", err)
+	}
+	re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")
```

Comment:
We can/should probably make this regex a static value, I think? It doesn't change between calls.

Reply:
I would extract this in `eth` backend (as Near address determining to `near` backend), but let's do it when we update to `Kwil 0.7`
We will do a lot of Go code writing and some refactoring for sure.